services/datamanager/builtin/builtin.go

```diff
@@ -512,16 +494,46 @@ func (svc *builtIn) uploadData(cancelCtx context.Context, intervalMins float64)
 			case <-ticker.C:
 				svc.lock.Lock()
 				if svc.syncer != nil {
-					svc.flushCollectors()
-					svc.syncer.SyncDirectory(svc.captureDir)
-					svc.syncAdditionalSyncPaths()
+					svc.sync()
 				}
 				svc.lock.Unlock()
 			}
 		}
 	})
 }
 
+func (svc *builtIn) sync() {
+	svc.flushCollectors()
+	toSync := getAllFilesToSync(svc.captureDir, svc.waitAfterLastModifiedMillis)
+	for _, ap := range svc.additionalSyncPaths {
+		toSync = append(toSync, getAllFilesToSync(ap, svc.waitAfterLastModifiedMillis)...)
```

Comment:
It looks like this is hard-coded to 10 seconds; how did we decide this? Or can we just not be confident that a file in additional paths won't be modified at any point, so this is our best guess (hasn't been touched in 10 seconds, so probably good to sync). 

Also, do I understand correctly that only the BYO files in the additional sync paths could have been modified? Or are there other non-capture files outside of additional sync paths?

Reply:
> Or can we just not be confident that a file in additional paths won't be modified at any point, so this is our best guess (hasn't been touched in 10 seconds, so probably good to sync).

Exactly this - nothing particularly intelligent

> only the BYO files in the additional sync paths could have been modified? Or are there other non-capture files outside of additional sync paths?

I'm not sure I understand the question. Re "are there other non-capture files outside of additional sync paths" I guess a user could hypothetically write non-capture files to the capture directory. These would be treated the same way as the files in additionalSyncPaths